### ISSUE TITLE
check_hadoop_dfs.pl missing blocks with replication factor support (was: Support for CDH5)

### ISSUE THREAD

**Issue by marcinzaremba** [harisekhon/nagios-plugins#47]
Few checks needs a little help. Examples are HBase table names (like hbase:meta in 0.98) or new lines in HDFS report. Is there any chance to add? I can contribute on this on a branch.

**Comment by harisekhon** [comment]
I know where I'll make the necessary updates for HBase table, won't take me long...

Please list which plugins need updates, if you can post the outputs from those plugins with -vvv will help to narrow down the required code changes as well.

Thanks

Hari

**Comment by harisekhon** [comment]
@marcinpraczko 

I don't want to have to go through and test every plugin against all the different versions as I don't have the infrastructure for this and given most have already been tested and updated for HDP 2.1 & 2.2 (Apache Hadoop 2.4.0 and 2.6.0) as the reference platform.

If you can list which plugins exactly you need updates for and which version of CDH 5 as well for it would help. I tested the HDFS report (I assumed you meant check_hadoop_dfs.pl is already updated for Hadoop 2.6.0).

Need more information.

**Comment by marcinzaremba** [comment]
Me referred as @marcinzaremba.

check_hadoop_dfs & CDH5: http://pastebin.com/w7P5qJEQ (same with check_hadoop_balance).
It seems like output differs in one line.

**Comment by harisekhon** [comment]
@marcinzaremba 

Sorry it autocompleted to another marcin who previously contributed a patch :)

There is no documentation around this new field "with replication factor" (typical Hadoop)... so I've added the count from there to missing blocks although this might be included in the other missing block number which may lead to a higher count, but at least it won't miss anything. It's a one line change to switch to ignoring it if we can confirm we only need the original Missing blocks count.

I've push this new version, please try and let me know.

Thanks

Hari

**Comment by marcinzaremba** [comment]
Almost perfect, but after _blocks_ word there is a space so I changed line accordingly:
`} elsif(/^Missing blocks\s*\(with replication factor\s\d+\):\s*(\d+)\s*$/i){" #line 176`
and it worked.

Instead of just putting space it is safer like in cases above place zero or more white chars IMHO.

**Comment by harisekhon** [comment]
@marcinzaremba 

Thanks I've just made that change.

You're welcome to raise github pull requests to provide patches yourself... in which case you'll get attribution as a contributor on Github. I've given credit to you in the commit comment I made in this case.

Let's close this ticket and create separate tickets for each plugin with example output so it's easier to track the fixes, I'll rename this one too to make it describe this dfs check enhancement.
